components/integrations/bots/add_bot/add_bot.jsx

```diff
@@ -278,7 +278,14 @@ export default class AddBot extends React.Component {
             }
         } else {
             const result = await this.props.actions.createBot(bot);
-            if (result) {
+            const usernameError = Utils.isValidUsername(bot.username);
```

Comment:
I'm confused here -- we're calling `isValidUsername` after calling `createBot`?

Reply:
Ah...  good catch and I will refactor.